internal/store/driver.go

```diff
@@ -909,6 +890,34 @@ func (d *Driver) calculateHTTPSEdgesFromIngress(edgeMap map[string]ingressv1alph
 	}
 }
 
+// retrieves the traffic policy for an ingress and falls back to the modSet policy if it doesn't exist
+func (d *Driver) getPolicyJSON(ingress *netv1.Ingress, modSet *ingressv1alpha1.NgrokModuleSet) (json.RawMessage, error) {
+	var err error
+	var policyJSON json.RawMessage
+
+	trafficPolicy, err := d.getNgrokTrafficPolicyForIngress(ingress)
+
+	if err != nil {
+		d.log.Error(err, "error getting ngrok traffic policy for ingress", "ingress", ingress)
+		return nil, err
+	}
+
+	if modSet.Modules.Policy != nil && trafficPolicy != nil {
+		return nil, fmt.Errorf("cannot have both a traffic policy and a moduleset policy on ingress: %s", ingress.Name)
+	}
+
+	if trafficPolicy != nil {
+		return trafficPolicy.Spec.Policy, nil
+	}
+
+	if policyJSON, err = json.Marshal(modSet.Modules.Policy); err != nil {
```

Comment:
Seems like `modSet.Modules.Policy` needs a nil check? It's legal to not have a policy set from either modset or policy-crd. Unless this behaves nicely in that scenario (not erroring, returning an empty byte slice).

Reply:
Yeah, and this aligns with `json.Marshal` call with `modSet.Modules.Policy` not having a `nil` check prior to the traffic policy considerations: https://github.com/ngrok/kubernetes-ingress-controller/pull/370/files#diff-9ece2bb7eeaf621263ba5abe1c62e28734da229e31faf7348e3aa4b260fc73aaL847.

However, you're on to something because in general this func is now getting a pointer mod set and this abstracted helper func should `nil` check `modSet` in general.